### PR TITLE
Loans to directors cross validation with directors report

### PR DESCRIPTION
**Pull request by amoriartyCH**
This PR is to implement the cross validation needed for the loans to directors resource with the directors report.

If a user has a directors report then in the loans to directors resource, they can only supply director names which were originally listed on the directors report.

-Extract duplicate logic into a DirectorValidator class and unit test it.
-Add validation messages where needed
-Implement new logic and change / create unit tests as needed.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/DirectorValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DirectorValidatorTest {
+
+    @Mock
+    DirectorService directorService;
```
Comment (BSaunders95) [inline src/test/java/uk/gov/companieshouse/api/accounts/validation/DirectorValidatorTest.java:32]: Check your access modifiers

File: src/test/java/uk/gov/companieshouse/api/accounts/service/impl/LoanServiceImplTest.java [hunk 2]
```diff
@@ -281,7 +280,7 @@ void getLoanNotFound() throws DataException {
         when(request.getRequestURI()).thenReturn(URI);
 
         LoanEntity loanEntity = null;
```
Comment (BSaunders95) [inline src/test/java/uk/gov/companieshouse/api/accounts/service/impl/LoanServiceImplTest.java:282]: Is this used any more?

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/LoanValidator.java [hunk 3]
```diff
@@ -37,4 +58,19 @@ private void validateLoanCalculation(LoanBreakdownResource loanBreakdown, Errors
             addError(errors, incorrectTotal, LOANS_BREAKDOWN_PATH_BALANCE_AT_PERIOD_END);
 		}
 	}
+
+	private void crossValidateDirectorNameDR(Loan loan, Transaction transaction,
+											 String companyAccountId, HttpServletRequest request, Errors errors) throws DataException {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/validation/LoanValidator.java:63]: Nitpick: formatting appears a little 'odd' here?

**Comment by amoriartyCH** [comment]
Fixed all suggested changes and solved any issues found!

**Review by ch-tstrickland — COMMENTED** [review]
First point:
The commit message needs more detail & that detail is partly in the PR summary.

**Review by BSaunders95 — APPROVED** [review]

**Review by ch-tstrickland — APPROVED** [review]

**Pull request merged**